scripts/testCompatibility.sh

```diff
@@ -23,136 +30,204 @@ function showHelp {
-  echo "There was an issue linking the luigi-core module"
-  exit 2
-fi
-if [ ! -f $EXAMPLE_NODE_MODULES/luigi-client/package.json ]; then
-  echo "There was an issue linking the luigi-client module"
-  exit 2
-fi
+  echoe "Installing selected Luigi example app"
+  cd $EXAMPLE_DIR
+  npm install
+}
+
+linkLuigi() {
+  echoe "Linking current Luigi to selected version"
+  # remove installed luigi versions and symlink with latest
+  rm -rf $EXAMPLE_NODE_MODULES/luigi*
+  ln -s $LUIGI_DIR/core/public $EXAMPLE_NODE_MODULES/luigi-core
+  ln -s $LUIGI_DIR/client $EXAMPLE_NODE_MODULES/luigi-client
+  ls -la $EXAMPLE_NODE_MODULES
+  ls $EXAMPLE_NODE_MODULES/luigi-core
+  ls $EXAMPLE_NODE_MODULES/luigi-client
+
+  if [ ! -f $EXAMPLE_NODE_MODULES/luigi-core/package.json ]; then
+    echoe "There was an issue linking the luigi-core module"
+    exit 2
+  fi
+  if [ ! -f $EXAMPLE_NODE_MODULES/luigi-client/package.json ]; then
+    echoe "There was an issue linking the luigi-client module"
+    exit 2
+  fi
+}
+
+bundleLuigi() {
+  echoe "Bundling example app"
+  cd $EXAMPLE_DIR
+  npm run build
+}
+
+verifyAndStartWebserver() {
+  WS=`command -v ws`
+  if [ ! -x $WS ] || [ "$WS" == "" ] ; then
+    echoe "Installing webserver"
+    npm i -g local-web-server
+  fi
+
+  killWebServer
+
+  echoe "Run webserver"
+  cd $EXAMPLE_DIR
+  (cd dist && ws --port 4200 --spa index.html &)
+}
+
+startE2eTestrunner() {
+  echoe "Starting e2e test headless"
+  cd $EXAMPLE_DIR
+  # start separately
+  npm run e2e:run
+
+  # Check and kill webserver
+  killWebServer
+}
+
+
+# Script
 
-echo "Bundling example app"
-npm run build
+LUIGI_DIR="${PWD}"
+LUIGI_FOLDERNAME="luigi-compatibility-testing"
+LUIGI_DIR_TESTING="$LUIGI_DIR/../$LUIGI_FOLDERNAME"
+EXAMPLE_DIR="$LUIGI_DIR_TESTING/core/examples/luigi-sample-angular"
+EXAMPLE_NODE_MODULES=$EXAMPLE_DIR/node_modules/@kyma-project
+TESTONLY=""
 
-echo "Installing and running spa webserver"
-npm i -g local-web-server
-killWebServer
-(cd dist && ws --port 4200 --spa index.html &)
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+    # commands with input value, shift 2
+    -t|--tag) TAG="$2"; shift 2;;
+    # validity check for argumenst with input value
+    -t|--tag) echoe "$1 requires an argument" >&2; exit 1;;
 
-echo "Starting e2e test headless"
-# start separately
-npm run e2e:run
+    # commands without input value, shift 1
+    -i|--install) INSTALL="true"; shift 1;;
+
+    -to|--test-only) TESTONLY="true"; shift 1;;
 
-# Check and kill webserver
-killWebServer
+    -*) echoe "unknown argument: $1" >&2; showHelp; exit 1;;
+    *) handle_argument "$1"; shift 1;;
+  esac
+done
+
+
+if [ "" == "$TESTONLY" ]; then
+  promptForTag
+  verifyInstallation
+  checkoutLuigiToTestfolder
+  linkLuigi
+  bundleLuigi
+  verifyAndStartWebserver
+  startE2eTestrunner
+fi # --test-only
+
+if [ "true" == "$TESTONLY" ]; then
+  echoe "Running bunded example and e2e tests"
+  verifyAndStartWebserver
+  startE2eTestrunner
```

Comment:
If `verifyAndStartWebserver` and `startE2eTestrunner` are included in both `if` maybe it make sense to move it out and use once.

Reply:
Yes, makes sense, I just explicitly put it there to make it readable easier.